x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java

```diff
@@ -115,14 +115,13 @@ public Collection<Object> createComponents(
         NamedWriteableRegistry namedWriteableRegistry,
         IndexNameExpressionResolver indexNameExpressionResolver,
         Supplier<RepositoriesService> repositoriesServiceSupplier,
-        Tracer tracer
+        Tracer tracer,
+        AllocationDeciders allocationDeciders
     ) {
         this.clusterServiceHolder.set(clusterService);
-        return List.of(
-            new AutoscalingCalculateCapacityService.Holder(this),
-            autoscalingLicenseChecker,
-            new AutoscalingNodeInfoService(clusterService, client)
-        );
+        var capacityService = new AutoscalingCalculateCapacityService(getDeciderServices());
```

Comment:
Main changes start here, with allocation deciders passed in, we don't seem to need the holder anymore and we can directly construct the capacity service.